src/libstd/sync/mutex.rs

```diff
@@ -205,10 +205,19 @@ impl<T: ?Sized> Mutex<T> {
     /// held. An RAII guard is returned to allow scoped unlock of the lock. When
     /// the guard goes out of scope, the mutex will be unlocked.
     ///
+    /// The exact behavior on locking a mutex twice in the same thread is left
```

Comment:
“locking a mutex twice” is ambiguous in whether 2nd lock happens with mutex being locked or unlocked.

I would reword this as “behaviour of locking the mutex in the thread which already holds the lock is left unspecified” instead.